Tests/libhostmgrTests/BuildkiteScriptBuilderTests.swift

```diff
@@ -0,0 +1,148 @@
+        let env = try DotEnv.read(path: codeQuoteEnvironmentPath)
+        scriptBuilder.copyEnvironmentVariables(prefixedBy: "BUILDKITE_", from: readLines(from: env))
+        XCTAssertEqual(
+            BuildkiteScriptBuilder.Value(wrapping: "automattic"),
+            scriptBuilder.environmentVariables["BUILDKITE_ORGANIZATION_SLUG"]
+        )
+        XCTAssertEqual(
+            "automattic",
+            scriptBuilder.environmentVariables["BUILDKITE_ORGANIZATION_SLUG"]?.escapedRepresentation
+        )
+    }
+
+    func testThatCommitMessageEnvironmentVariableIsImported() throws {
+        let env = try DotEnv.read(path: codeQuoteEnvironmentPath)
+        scriptBuilder.copyEnvironmentVariables(prefixedBy: "BUILDKITE_", from: readLines(from: env))
+        XCTAssertEqual(
+            BuildkiteScriptBuilder.Value(wrapping: "A simple message with `code quotes`"),
+            scriptBuilder.environmentVariables["BUILDKITE_MESSAGE"]
+        )
+        XCTAssertEqual(
+            "A simple message with \\`code quotes\\`",
+            scriptBuilder.environmentVariables["BUILDKITE_MESSAGE"]?.escapedRepresentation
+        )
+    }
+
+    func testThatPullRequestEnvironmentVariableIsImported() throws {
+        let env = try DotEnv.read(path: codeQuoteEnvironmentPath)
+        scriptBuilder.copyEnvironmentVariables(prefixedBy: "BUILDKITE_", from: readLines(from: env))
+        XCTAssertEqual(
+            BuildkiteScriptBuilder.Value(wrapping: "19136"),
+            scriptBuilder.environmentVariables["BUILDKITE_PULL_REQUEST"]
+        )
+    }
+
+    func testThatCommitMessageWithCodeQuotesIsProperlyEscaped() throws {
+        let original = try getContentsOfResourceAsValue(
+            named: "buildkite-commit-message-original",
+            withExtension: "txt"
+        )
+
+        XCTAssertEqual(
+            try getContentsOfResource(named: "buildkite-commit-message-expected", withExtension: "txt"),
+            original.escapedRepresentation
+        )
+    }
+
+    // MARK: - Command Tests
+    func testThatCommandEscapingDoesNotEscapeMultiWordCommands() throws {
+        let command = BuildkiteScriptBuilder.Command(command: "buildkite-agent bootstrap")
+        XCTAssertEqual("buildkite-agent bootstrap", command.escapedText)
+    }
+
+    func testThatCommandDoesNotEscapeCompoundCommands() throws {
+        let command = BuildkiteScriptBuilder.Command(command: "buildkite-agent", arguments: ["bootstrap"])
+        XCTAssertEqual("buildkite-agent bootstrap", command.escapedText)
+    }
+
+    func testThatCommandEscapesSpacesInArguments() throws {
+        let command = BuildkiteScriptBuilder.Command(
+            command: "buildkite-agent bootstrap",
+            arguments: ["/Users/my builder user/.bashrc"]
+        )
+
+        // Note: this also tests that escaped strings are wrapped in quotes
+        XCTAssertEqual("buildkite-agent bootstrap \"/Users/my\\ builder\\ user/.bashrc\"", command.escapedText)
```

Comment:
My first reading of this made me think "wait the quote before the path shouldn't be escaped by the script builder here"… before I realized the escaping was for swift itself. I suggest using raw string syntax to avoid that confusion here.
```suggestion
        XCTAssertEqual(#"buildkite-agent bootstrap "/Users/my\\ builder\\ user/.bashrc""#, command.escapedText)
```

Reply:
Also, is it intended that we both quote _and_ escape the spaces here? Wouldn't that make then double-escaping?

Reply:
mmm yep, that is indeed double-escaped, which isn't what we're after. Addressed in 5aa2ed2abe1a1980890b3030398a3482c2d0b241.